rollup.config.js

```diff
@@ -1,74 +1,77 @@
-import esbuild from 'rollup-plugin-esbuild';
+import typescript from '@rollup/plugin-typescript';
 import copy from 'rollup-plugin-copy';
-import dts from 'rollup-plugin-dts';
+import esbuild from 'rollup-plugin-esbuild';
 import replace from 'rollup-plugin-replace';
-import packageJson from './package.json';
 
 const NODE_ENV = process.env.NODE_ENV || 'development';
 
-const bundle = config => [
-  {
-    ...config,
-    external: id => !/^[./]/.test(id),
-    plugins: [
-      ...config.plugins,
-      replace({
-        'process.env.NODE_ENV': JSON.stringify(NODE_ENV)
-      }),
-      esbuild({ loaders: { '.js': 'jsx' } })
-    ]
-  },
-  {
-    ...config,
-    output: config.output.dir
-      ? config.output
-      : {
-          file: packageJson.types,
-          format: 'es'
-        },
-    external: id => !/^[./]/.test(id),
-    plugins: [
-      ...config.plugins,
-      replace({
-        'process.env.NODE_ENV': JSON.stringify(NODE_ENV)
-      }),
-      dts()
-    ]
-  }
-];
+const baseBundle = {
+  external: id => !/^[./]/.test(id),
+  plugins: [
+    replace({
+      'process.env.NODE_ENV': JSON.stringify(NODE_ENV)
+    }),
+    esbuild()
+  ]
+};
 
 export default [
-  ...bundle({
-    input: './src/index.ts',
+  {
+    ...baseBundle,
+    input: ['./src/index.ts', './src/types.tsx'],
```

Comment:
@arturbien @luizbaldi apparently types breaking on the consumers was caused by `types` not being exported.